effects: link the effect modules against effects_base

builtin, builtin_collection, extensions, nyquist and vst use classes from
effects_base (Au3EffectLoader, Au3AudioPluginScanner, AbstractViewLauncher,
...) without declaring it. lld and the MSVC linker resolve that from the
static libraries in any order, GNU ld does not (undefined references with
GCC). The lv2 module already declares it.

diff --git a/src/effects/builtin/CMakeLists.txt b/src/effects/builtin/CMakeLists.txt
index c1b434b..a6b6037 100644
--- a/src/effects/builtin/CMakeLists.txt
+++ b/src/effects/builtin/CMakeLists.txt
@@ -37,7 +37,7 @@ include(${CMAKE_CURRENT_LIST_DIR}/../../au3wrap/au3wrapDefs.cmake)
 set(MODULE_INCLUDE ${AU3_INCLUDE})
 set(MODULE_DEF ${AU3_DEF})
 
-set(MODULE_LINK au3wrap)
+set(MODULE_LINK effects_base au3wrap)
 
 set(MODULE_USE_UNITY OFF)
 
diff --git a/src/effects/builtin_collection/CMakeLists.txt b/src/effects/builtin_collection/CMakeLists.txt
index 6bf1fda..7399940 100644
--- a/src/effects/builtin_collection/CMakeLists.txt
+++ b/src/effects/builtin_collection/CMakeLists.txt
@@ -191,7 +191,7 @@ set(MODULE_DEF ${MODULE_DEF}
     # -DNOTE_TRACK_API=
 )
 
-set(MODULE_LINK au3wrap libsbsms shared)
+set(MODULE_LINK effects_base au3wrap libsbsms shared)
 set(MODULE_LINK_PUBLIC au3-effects au3-builtin-effects)
 
 set(MODULE_USE_UNITY OFF)
diff --git a/src/effects/extensions/CMakeLists.txt b/src/effects/extensions/CMakeLists.txt
index cfdb643..37f6d7e 100644
--- a/src/effects/extensions/CMakeLists.txt
+++ b/src/effects/extensions/CMakeLists.txt
@@ -33,7 +33,7 @@ include(${CMAKE_CURRENT_LIST_DIR}/../../au3wrap/au3wrapDefs.cmake)
 
 set(MODULE_INCLUDE ${AU3_INCLUDE})
 set(MODULE_DEF ${AU3_DEF})
-set(MODULE_LINK au3wrap trackedit muse::extensions)
+set(MODULE_LINK effects_base au3wrap trackedit muse::extensions)
 set(MODULE_USE_UNITY OFF)
 
 setup_module()
diff --git a/src/effects/nyquist/CMakeLists.txt b/src/effects/nyquist/CMakeLists.txt
index 8749201..2c87c07 100644
--- a/src/effects/nyquist/CMakeLists.txt
+++ b/src/effects/nyquist/CMakeLists.txt
@@ -36,7 +36,7 @@ include(${CMAKE_CURRENT_LIST_DIR}/../../au3wrap/au3wrapDefs.cmake)
 set(MODULE_INCLUDE ${AU3_INCLUDE})
 set(MODULE_DEF ${AU3_DEF})
 
-set(MODULE_LINK au3wrap effects_builtin au3-nyquist-effects)
+set(MODULE_LINK effects_base au3wrap effects_builtin au3-nyquist-effects)
 
 add_subdirectory(libnyquist)
 
diff --git a/src/effects/vst/CMakeLists.txt b/src/effects/vst/CMakeLists.txt
index 5e47832..a30f59b 100644
--- a/src/effects/vst/CMakeLists.txt
+++ b/src/effects/vst/CMakeLists.txt
@@ -38,6 +38,7 @@ set(MODULE_INCLUDE ${AU3_INCLUDE} ${MUSE_FRAMEWORK_SRC_PATH}/vst)
 set(MODULE_DEF ${AU3_DEF})
 
 set(MODULE_LINK
+        effects_base
         au3wrap
         vst_sdk_3 # Note: vst_sdk_3 needed because au3-vst3 public headers expose VST3 SDK types
 )
